src/canvas/DOMManagers/DOMManager.ts

```diff
@@ -0,0 +1,100 @@
+import { getEnv } from '../../env';
+import { TSize } from '../../typedefs';
+import {
+  CSSDimensions,
+  getElementOffset,
+  setCSSDimensions,
+  setCanvasDimensions,
+} from './util';
+
+export type CanvasItem = {
+  el: HTMLCanvasElement;
+  ctx: CanvasRenderingContext2D;
+};
+
+export type ItemMap<T = HTMLCanvasElement> =
+  | {
+      main: T;
+      [x: string]: T;
+    }
+  | {
+      main: T;
+      top: T;
+      [x: string]: T;
+    };
+
+export type StaticDOMManagerType = DOMManager<{
+  main: HTMLCanvasElement;
+  [x: string]: HTMLCanvasElement;
+}>;
+export type DOMManagerType = DOMManager<{
+  main: HTMLCanvasElement;
+  top: HTMLCanvasElement;
+  [x: string]: HTMLCanvasElement;
+}>;
+
+export class DOMManager<T extends ItemMap> {
+  readonly items: Record<keyof T, CanvasItem>;
+  readonly container?: HTMLDivElement;
+
+  static build(
+    main: HTMLCanvasElement,
+    top?: HTMLCanvasElement,
+    container?: HTMLDivElement
+  ) {
+    return new this(
+      top
+        ? {
+            main,
+            top,
+          }
+        : { main },
+      container
+    );
+  }
+
+  constructor(items: T, container?: HTMLDivElement) {
+    this.items = Object.fromEntries(
+      Object.entries(items).map(([key, canvas]) => [
+        key,
+        { el: canvas, ctx: canvas.getContext('2d')! },
+      ])
+    ) as Record<keyof T, CanvasItem>;
+    this.container = container;
+  }
+
+  setDimensions(size: TSize, retinaScaling: number) {
+    Object.values(this.items).forEach(({ el, ctx }) =>
+      setCanvasDimensions(el, ctx, size, retinaScaling)
+    );
+  }
+
+  setCSSDimensions(size: Partial<CSSDimensions>) {
+    Object.values(this.items).forEach(({ el }) => setCSSDimensions(el, size));
+    this.container && setCSSDimensions(this.container, size);
```

Comment:
hmm... I have thoughts about this.
Is this legacy only? Is there a case you want to resize the container?

Reply:
You want to resize the container every time you want a different css size compared to the one that are given initially by the ration between screen pixels and css pixels.
When you want to enforce a different scaling that window pixel ratio

Reply:
I am trying to remove it by not setting `position: absolute` on the main canvas element